lib/private/Security/Crypto.php

```diff
@@ -114,6 +114,18 @@ private function decryptWithoutSecret(string $authenticatedCiphertext, string $p
 			throw new Exception('Authenticated ciphertext could not be decoded.');
 		}
 
+		// Rearrange arguments for legacy ownCloud migrations
+		$originalParts = $parts;
+		if ($partCount === 4 && $parts[0] === 'v2') {
```

Comment:
*nit* I would save this condition as a variable, so it's clearer that this is the legacy mode

```suggestion
		$legacyOwnCloudMigration = $partCount === 4 && $parts[0] === 'v2'
		if ($legacyOwnCloudMigration) {
```